odb_extractor_atoms.py

```diff
@@ -6,15 +6,179 @@
 # 2021-11-24
 
 import os
+import sys
 import json
 import gzip
 import numpy as np
+import matplotlib.pyplot as plt
+from scipy.signal import find_peaks
+import requests
 
 from openpyxl import Workbook
 from openpyxl import load_workbook
 
 
-fpuinst = {'GNIRS': 'instrument:slitWidth', 'GMOS-N': 'instrument:fpu'}
+fpuinst = {'GSAOI': 'instrument:utilityWheel', 'GPI': 'instrument:observingMode', 'Flamingos2': 'instrument:fpu',
+           'NIFS': 'instrument:mask', 'GNIRS': 'instrument:slitWidth', 'GMOS-N': 'instrument:fpu',
+           'GMOS-S': 'instrument:fpu', 'NIRI': 'instrument:mask'}
+
+gpi_filter_wav = {'Y': 1.05, 'J': 1.25, 'H': 1.65, 'K1': 2.05, 'K2': 2.25}
+nifs_filter_wav = {'ZJ': 1.05, 'JH': 1.25, 'HK': 2.20}
+
+
+def find_filter(input, filter_dict):
+    """Match input string with filter list (in dictionary)"""
+
+    filter = ''
+    filters = list(filter_dict.keys())
+    for filt in filters:
+        if filt in input:
+            filter = filt
+            break
+    return filter
```

Comment:
These are things that @stroncod or I can change in the implementation if necessary, but this implementation would be more efficient, and I would think we would want to return `None` instead of the empty string if nothing was found? (If not, we can change the `None` to the empty string in the second line and remove the `Optional` type hint.)

```
def find_filter(input: str, filter_dict: Mapping[str, str]) -> Optional[str]:
    return next(filter(lambda f: f in input, filter_dict), None)
```